src/rtutils.c

```diff
@@ -968,22 +975,33 @@ static size_t jl_static_show_x_(JL_STREAM *out, jl_value_t *v, jl_datatype_t *vt
     else if (jl_datatype_type && jl_is_datatype(vt)) {
         int istuple = jl_is_tuple_type(vt), isnamedtuple = jl_is_namedtuple_type(vt);
         size_t tlen = jl_datatype_nfields(vt);
-        if (isnamedtuple) {
-            if (tlen == 0)
-                n += jl_printf(out, "NamedTuple");
-        }
-        else if (!istuple) {
-            n += jl_static_show_x(out, (jl_value_t*)vt, depth);
-        }
-        n += jl_printf(out, "(");
         size_t nb = jl_datatype_size(vt);
         if (nb > 0 && tlen == 0) {
             uint8_t *data = (uint8_t*)v;
-            n += jl_printf(out, "0x");
-            for(int i = nb - 1; i >= 0; --i)
-                n += jl_printf(out, "%02" PRIx8, data[i]);
-        }
-        else {
+            n += jl_printf(out, "reinterpret(");
+            n += jl_static_show_x(out, (jl_value_t*)vt, depth);
+            n += jl_printf(out, ", 0x");
+            uint32_t a = 0x01020304;
+            uint8_t *endian_bom = (uint8_t*) &a;
```

Comment:
this is undefined behavior (since it is observably different on different platforms). use the standard macro (`BYTE_ORDER`)